### PR TITLE
filter reset code added

### PR DESCRIPTION
**Pull request by rakeshSgr**

### CODE REVIEW

File: projects/sb-dashlets/src/lib/dashlet.module.ts [hunk 1]
```diff
@@ -26,7 +28,7 @@ export class DashletModule {
     return {
       ngModule: DashletModule,
       providers: [
-        { provide: DATA_SERVICE, useClass: (config && config.dataService) || DataService }
+        { provide: DATA_SERVICE, useClass: DataService }
```
Comment (yravinderkumar33) [inline projects/sb-dashlets/src/lib/dashlet.module.ts]: Why was it changed ?

Comment (yravinderkumar33) [reply projects/sb-dashlets/src/lib/dashlet.module.ts]: It enables the client to send their own implementation of data service. If no provided then the default one is used

File: projects/sb-dashlets/src/lib/components/big-number/big-number.component.html [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 <ng-template #defaultFilterTemplate let-context>
   <sb-dashlets-filters [data]="context?.data" [config]="context?.config?.filters"
-    (filteredData)="update({data: $event})">
+    (filteredData)="update({data: $event})" #filterDom>
```
Comment (yravinderkumar33) [inline projects/sb-dashlets/src/lib/components/big-number/big-number.component.html]: This implementation will not handle the generalization for the filters. Please revamp acoordingly.

File: projects/sb-dashlets/src/lib/components/base/base.component.ts [hunk 3]
```diff
@@ -19,6 +19,8 @@ export abstract class BaseComponent implements Partial<IBase> {
 
   state = new EventEmitter<ReportState>();
   events = new EventEmitter<CustomEvent>();
+  eventsSubject: Subject<void> = new Subject<void>();
+
   x
```
Comment (yravinderkumar33) [inline projects/sb-dashlets/src/lib/components/base/base.component.ts]: Please remove this x

File: projects/sb-dashlets/src/lib/components/dashlet/dashlet.component.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ export class DashletComponent implements OnInit {
   @Input() data: object;
 
   @Output() events = new EventEmitter();
+  @Output() filterEvents = new EventEmitter();
```
Comment (yravinderkumar33) [inline projects/sb-dashlets/src/lib/components/dashlet/dashlet.component.ts]: This piece of code is not being used. Please remove it.

Comment (rakeshSgr) [reply projects/sb-dashlets/src/lib/components/dashlet/dashlet.component.ts]: resolved

File: projects/sb-dashlets/src/lib/components/base/base.component.ts (hunk 3 above)
Comment (rakeshSgr) [reply projects/sb-dashlets/src/lib/components/base/base.component.ts]: done

**Review by yravinderkumar33 — APPROVED** [review]

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**